hnn_core/cell.py

```diff
@@ -25,6 +25,68 @@ def _get_cos_theta(p_secs, sec_name_apical):
     return cos_thetas
 
 
+def _calculate_gaussian(x_val, height, lamtha):
+    """Return height of gaussian at x_val.
+
+    Parameters
+    ----------
+    x_val : float
+        Value on x-axis to query height of gaussian curve.
+    height : float
+        Height of the gaussian curve at zero.
+    lamtha : float
+        Space constant.
+
+    Returns
+    -------
+    x_height : float
+        Height of gaussian at x_val.
+
+    Notes
+    -----
+    Gaussian curve is centered at zero and has a fixed peak height
+    such the _calculate_gaussian(0, lamtha) returns 1 for all lamtha.
+    """
+    x_height = height * np.exp(-(x_val**2) / (lamtha**2))
+
+    return x_height
+
+
+def _get_gaussian_connection(src_pos, target_pos, nc_dict):
+    """Calculate distance dependent connection properties.
+
+    Parameters
+    ----------
+    src_pos : float
+        Position of source cell.
+    target_pos : float
+        Position of target cell.
+    nc_dict : dict
+        Dictionary with keys: pos_src, A_weight, A_delay, lamtha
+        Defines the connection parameters
+
+    Returns
+    -------
+    weight : float
+        Weight of the synaptic connection.
+    delay : float
+        Delay of synaptic connection.
+
+    Notes
+    -----
+    Distance in xy plane is used for gaussian decay.
+    """
+    x_dist = target_pos[0] - src_pos[0]
+    y_dist = target_pos[1] - src_pos[1]
+    cell_dist = np.sqrt(x_dist**2 + y_dist**2)
```

Comment:
Shouldn't this currently be XZ-plane?! Or is the assumption that the cells are already reoriented to have apicals in Z? In any case, having this hard-coded requires us to define our coordinate system, soon...

Reply:
This function is called from `Network`, which implements an abstract model that distributes cells across the XY-plane. To your point, though, this needs to be resolved soon because we keep tripping over the disparity between the abstract network model and the low-level hoc Neuron model.

Reply:
Right, I see.